eips_code/eip-4750.py

```diff
@@ -0,0 +1,445 @@
+# The ranges below are as specified in the Yellow Paper.
+# Note: range(s, e) excludes e, hence the +1
+valid_opcodes = [
+    *range(0x00, 0x0b + 1),
+    *range(0x10, 0x1d + 1),
+    0x20,
+    *range(0x30, 0x3f + 1),
+    *range(0x40, 0x48 + 1),
+    *range(0x50, 0x5f + 1),
+    *range(0x60, 0x6f + 1),
+    *range(0x70, 0x7f + 1),
+    *range(0x80, 0x8f + 1),
+    *range(0x90, 0x9f + 1),
+    *range(0xa0, 0xa4 + 1),
+    # Note: 0xfe is considered assigned.
+    *range(0xf0, 0xf5 + 1), 0xfa, 0xfd, 0xfe, 0xff
+]
+
+# STOP, RETF, RETURN, REVERT, INVALID, SELFDESTRUCT
+terminating_opcodes = [ 0x00, 0x5f, 0xf3, 0xfd, 0xfe, 0xff ]
+
+immediate_sizes = 256 * [0]
+immediate_sizes[0x5c] = 2  # RJUMP
+immediate_sizes[0x5d] = 2  # RJUMPI
+immediate_sizes[0x5e] = 2  # CALLF
+for opcode in range(0x60, 0x7f + 1):  # PUSH1..PUSH32
+    immediate_sizes[opcode] = opcode - 0x60 + 1
+
+# Fails with assertion on invalid code
```

Comment:
I think we should drop this way of validating. This causes more troubles than needed.

Reply:
What troubles?

Reply:
1. You need to wrap these functions to return bool to test them.
2. You cannot distinguish real asserts (e.g. for already validated preconditions) from validation.
3. You can have error codes for better testing: to make sure you are really testing given validation rule and failure is not coming from some earlier/later rule.

Reply:
I think the argument for it was to present most simple and readable code inside EIPs.